### ISSUE TITLE
Can't "Use node.range[0] instead of node.start"

### ISSUE THREAD

**Issue by Skalman** [eslint/eslint#13610]
**Tell us about your environment**

Node version: v12.10.0
npm version: v6.10.3
ESLint version: v7.7.0

**What parser are you using?**
default

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

I'm creating a rule that might replace parts of a `TemplateElement`. I want to replace the raw value with a new value, and it seems like `node.start` and `node.end` correspond to the range of the raw value, while `node.range` corresponds to more than the "raw" value.

<details>
<summary>Examples showing why I need access to <code>start</code> and <code>end</code></summary>

```js
// EXAMPLE 1
// Start/end match the value:
`a`
 ^

// Range matches the whole template string:
`a`
^^^

// Relevant AST part:
{
  "type": "TemplateElement",
  "start": 1,
  "end": 2,
  "range": [0, 3],
  "value": { "raw": "a", "cooked": "a" },
  "tail": true
}
```

```js
// EXAMPLE 2
// Start/end match the value:
`a${a}`
 ^

// Range matches surrounding parts of the template string:
`a${a}`
^^^^

// Relevant AST part:
{
  "type": "TemplateElement",
  "start": 1,
  "end": 2,
  "range": [0, 4],
  "value": { "raw": "a", "cooked": "a" },
  "tail": false
}
```
</details>

So in my rule, I'm doing this:

```js
context.report({
  node,
  messageId: "x",
  fix(fixer) {
    return fixer.replaceTextRange([node.start, node.end], fixedValue);
  },
});
```

And in my rule's test:

```js
ruleTester.run("x", myRule, {
  valid: ["`a`"],
  invalid: [{ code: "`b`", errors: 1 }],
});
```

**What did you expect to happen?**

I expected to be able to access `node.start` and `node.end`.

**What actually happened? Please include the actual, raw output from ESLint.**

The RuleTester's parser throws an error `Use node.range[0] instead of node.start`.

```
$ npm test

> eslint-plugin-my-plugin@0.1.0 test /path/to/eslint-plugin-my-plugin
> mocha tests --recursive



  x
    valid
      ✓ `a` (52ms)
    invalid
      1) `b`


  1 passing (69ms)
  1 failing

  1) x
       invalid
         `b`:
     Error: Use node.range[0] instead of node.start
Occurred while linting <input>:1
      at Node.get (node_modules/eslint/lib/rule-tester/rule-tester.js:223:23)
      at Object.fix (lib/rules/x.js:59:47)
      at normalizeFixes (node_modules/eslint/lib/linter/report-translator.js:178:28)
      at /path/to/eslint-plugin-my-plugin/node_modules/eslint/lib/linter/report-translator.js:343:49
      at Object.report (node_modules/eslint/lib/linter/linter.js:917:41)
      at process (lib/rules/x.js:49:15)
      at TemplateElement (lib/rules/x.js:66:34)
      at /path/to/eslint-plugin-my-plugin/node_modules/eslint/lib/linter/safe-emitter.js:45:58
      at Array.forEach (<anonymous>)
      at Object.emit (node_modules/eslint/lib/linter/safe-emitter.js:45:38)
      at NodeEventGenerator.applySelector (node_modules/eslint/lib/linter/node-event-generator.js:254:26)
      at NodeEventGenerator.applySelectors (node_modules/eslint/lib/linter/node-event-generator.js:283:22)
      at NodeEventGenerator.enterNode (node_modules/eslint/lib/linter/node-event-generator.js:297:14)
      at CodePathAnalyzer.enterNode (node_modules/eslint/lib/linter/code-path-analysis/code-path-analyzer.js:673:23)
      at /path/to/eslint-plugin-my-plugin/node_modules/eslint/lib/linter/linter.js:949:32
      at Array.forEach (<anonymous>)
      at runRules (node_modules/eslint/lib/linter/linter.js:944:15)
      at Linter._verifyWithoutProcessors (node_modules/eslint/lib/linter/linter.js:1170:31)
      at Linter.verify (node_modules/eslint/lib/linter/linter.js:1233:21)
      at runRuleForItem (node_modules/eslint/lib/rule-tester/rule-tester.js:556:37)
      at testInvalidTemplate (node_modules/eslint/lib/rule-tester/rule-tester.js:654:28)
      at Context.<anonymous> (node_modules/eslint/lib/rule-tester/rule-tester.js:893:25)
      at processImmediate (internal/timers.js:439:21)



npm ERR! Test failed.  See above for more details.
```

**Are you willing to submit a pull request to fix this bug?**

Yes, either by excluding `TemplateElement`s or by excluding all nodes where the `start` and `end` differ from the `range`.
